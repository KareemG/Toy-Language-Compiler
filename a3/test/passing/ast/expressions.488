{
    var a : integer
    var b, c, d, e : integer
    var f, g, h : boolean

    read a, b, c

    d := -(a + b) * a + b * c
    e := a - -(b / d)

    f := d < e
    g := not (d * 2 > e)
    h := ((d*d + e*e < 10000) ? false : true)

    if h then write "less than 10000"

    if d not = e then write "not equal"
    if d <  e then write "<"
    if d <= e then write "<="
    if d >  e then write ">"
    if d >= e then write ">=" 

    if f and g then write "f and g"
    if f or g then write "f or g"
    if not f and g then write "not f and g"
    if f or not g then write "f or not g"
}