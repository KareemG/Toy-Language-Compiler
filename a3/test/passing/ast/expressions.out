 { 
 var a : integer
 var b, c, d, e : integer
 var f, g, h : boolean
 read a, b, c
 d : = ((-((a) + (b))) * (a)) + ((b) * (c))
 e : = (a) - (-((b) / (d)))
 f : = (d) < (e)
 g : = not(((d) * (2)) > (e))
 h : = ((((d) * (d)) + ((e) * (e))) < (10000)?false:true)
 if h then
 write "less than 10000"end
 if (d) not = (e) then
 write "not equal"end
 if (d) < (e) then
 write "<"end
 if (d) <= (e) then
 write "<="end
 if (d) > (e) then
 write ">"end
 if (d) >= (e) then
 write ">="end
 if (f) and (g) then
 write "f and g"end
 if (f) or (g) then
 write "f or g"end
 if (not(f)) and (g) then
 write "not f and g"end
 if (f) or (not(g)) then
 write "f or not g"end
 }
